### PR TITLE
Fix paramiko options

### LINKED ISSUE #52103: ansible_paramiko_user is ignored
##### SUMMARY
The docs for [paramiko_ssh](https://docs.ansible.com/ansible/latest/plugins/connection/paramiko_ssh.html) state that `remote_user` can be set using the var: `ansible_paramiko_user`or environment variable `ANSIBLE_PARAMIKO_REMOTE_USER`. However, these are ignored, even when `ansible_connection=paramiko` is present.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
`_play_context` setup for connection plugins (probably in `playbook/play_context.py`.

##### ANSIBLE VERSION
```
$ ansible --version
ansible 2.7.6
  config file = None
  configured module search path = ['/Users/jdashton/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/Cellar/ansible/2.7.6/libexec/lib/python3.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.7.2 (default, Feb 12 2019, 08:15:36) [Clang 10.0.0 (clang-1000.11.45.5)]
$
```

##### CONFIGURATION
```
$ ansible-config dump --only-changed
$
```

##### OS / ENVIRONMENT
macOS Mojave Version 10.14.4 Beta (18E184e)

##### DEPENDENCY PROBLEMS
```paste below

```

##### STEPS TO REPRODUCE
1. In an empty directory, cat the following to `hosts-test`:
```
my-host  ansible_host=192.168.162.101 ansible_paramiko_user=my_user ansible_connection=paramiko
```
2. Run this command:
```
ansible all -i hosts-test -m ping -vvv
```

##### EXPECTED RESULTS
Output should include this line with the `my_user` remote user:
```
<192.168.162.101> ESTABLISH PARAMIKO SSH CONNECTION FOR USER: my_user on PORT 22 TO 192.168.162.101
```

##### ACTUAL RESULTS
Output includes this line with `None` as the remote user:
```
<192.168.162.101> ESTABLISH PARAMIKO SSH CONNECTION FOR USER: None on PORT 22 TO 192.168.162.101
```

### PR DESCRIPTION
**Pull request by bcoca**
fixes #52103

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

paramiko_ssh

### CODE REVIEW

**Comment by trishnaguha** [comment]
cc @Qalthos

File: lib/ansible/plugins/doc_fragments/connection_persistent.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+class ModuleDocFragment(object):
+
+    # Standard files documentation fragment
+    DOCUMENTATION = r'''
+options:
+  persistent_connect_timeout:
```
Comment (Qalthos) [inline lib/ansible/plugins/doc_fragments/connection_persistent.py:13]: If the goal is to make this reusable in other persistent connections, that sounds fine, but `persistent_connect_timeout` doesn't need to be in every plugin (`httpapi` in particular has no independent connect step to time out from) and should probably be returned to `network_cli`

File: lib/ansible/plugins/doc_fragments/connection_persistent.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        close.
+    default: 30
+    ini:
+      - section: persistent_connection
+        key: command_timeout
+    env:
+      - name: ANSIBLE_PERSISTENT_COMMAND_TIMEOUT
+    vars:
+      - name: ansible_command_timeout
+  persistent_buffer_read_timeout:
```
Comment (Qalthos) [inline lib/ansible/plugins/doc_fragments/connection_persistent.py]: Similarly, this option is highly `network_cli`-specific, so much that I'm pretty sure `netconf` doesn't even need it. This should go back to `network_cli`

**Review by Qalthos — CHANGES_REQUESTED** [review]

File: lib/ansible/plugins/doc_fragments/connection_persistent.py (hunk 2 above)
Comment (Qalthos) [reply lib/ansible/plugins/doc_fragments/connection_persistent.py]: I was asking for `persistent_buffer_read_timeout` to be returned to `network_cli`, and _not_ `persistent_command_timeout`.

**Review by Qalthos — CHANGES_REQUESTED** [review]

**Review by Qalthos — APPROVED** [review]

**Comment by samdoran** [comment]
!waffling ci_verified

File: lib/ansible/plugins/doc_fragments/connection_persistent.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+class ModuleDocFragment(object):
+
+    # Standard files documentation fragment
+    DOCUMENTATION = r'''
+options:
```
Comment (Qalthos) [inline lib/ansible/plugins/doc_fragments/connection_persistent.py:12]:
```suggestion
options:
  persistent_connect_timeout:
    type: int
    description:
    - Configures, in seconds, the amount of time to wait when trying to initially
      establish a persistent connection.  If this value expires before the connection
      to the remote device is completed, the connection will fail.
    default: 30
    ini:
    - section: persistent_connection
      key: connect_timeout
    env:
    - name: ANSIBLE_PERSISTENT_CONNECT_TIMEOUT
    vars:
    - name: ansible_connect_timeout
```

**Review by Qalthos — CHANGES_REQUESTED** [review]
Not sure what the current status on this is, but if this happens, the persistent doc fragments should have connect_timeout as well

File: lib/ansible/plugins/doc_fragments/connection_persistent.py (hunk 3 above)
Comment (Qalthos) [reply lib/ansible/plugins/doc_fragments/connection_persistent.py:12]: Actually, on the whole, this belongs in netcommon, so now it will be: https://github.com/ansible-collections/ansible.netcommon/pull/308/files#diff-e84f32f4c9057b4900c0593346d486cf66db31cb92dcf9016d75d0bba03ce88d

Comment (bcoca) [reply lib/ansible/plugins/doc_fragments/connection_persistent.py:12]: ok, so no need to add this here now? have the rest of the PRs been merged and this will pass your tests now?

Comment (Qalthos) [reply lib/ansible/plugins/doc_fragments/connection_persistent.py:12]: Having had some time to reflect on it, this should probably stay here and get updated while `module_utils/connection.py` and friends are in core.

We want to maintain the ability to make persistent connections that don't depend on `ansible.netcommon`. `httpapi` for example could theoretically live somewhere else and not have much issue. A complete version of this doc_fragment has been merged to netcommon, but that really is only because this is pretty useful but we would not be able to use it for a long time if it were just in core.

Basically, my opinion is that while it remains possible to make a standalone persistent connection without any collections, (which is pretty much just needs `ansible-connection` & `connection.py`), then this doc_fragment should probably live here, too.

Comment (bcoca) [reply lib/ansible/plugins/doc_fragments/connection_persistent.py:12]: added

**Comment by Akasurde** [comment]
superseded by https://github.com/ansible/ansible/pull/79704
